polyfills/Map/polyfill.js

```diff
@@ -141,6 +141,9 @@
 
 	Map.length = 0;
 
+	Map.prototype.constructor = Map;
+	Map.name = "Map";
```

Comment:
The value of this comes from `Function.prototype.name`, which I think we already have a polyfill for.